lighthouse-cli/cli-flags.js

```diff
@@ -171,6 +184,24 @@ function getFlags(manualArgv) {
           'For more information on Lighthouse, see https://developers.google.com/web/tools/lighthouse/.')
       .wrap(yargs.terminalWidth())
       .argv;
+
+  // ".middleware" does not exist in this version of yargs, so do some post-processing here.
+  /** @type {Array<keyof LH.CliFlags>} */
+  const arrayKeysThatSupportCsv = [
+    'onlyAudits',
+    'onlyCategories',
+    'output',
+    'plugins',
+    'skipAudits',
+  ];
+  arrayKeysThatSupportCsv.forEach(key => {
+    const input = /** @type {string[]} */ (argv[key]);
+    if (input) {
+      argv[key] = flatten(input.map(value => value.split(',')));
```

Comment:
it's quite concerning this (`Array<string>`) is assignable back to `argv[key]`, which should be something like `number`|`boolean`|`string`|`Array<string>` not to mention our enums, etc...

The values are all there, it's just not flagging it...

Reply:
also, `[].flatMap`, coming in Node 11 :)

Reply:
> it's quite concerning this (Array<string>) is assignable back to argv[key], which should be something like number|boolean|string|Array<string> not to mention our enums, etc...

apparently this is a long-standing issue that will supposedly be fixed in https://github.com/microsoft/TypeScript/pull/30769

...but we can't run the current nightlies because they OOM on our code base, so I don't know for sure :)

We implicitly rely on this behavior in a few places in our codebase, so it'll be interesting to see how much breaks vs how much is still fine (last time I looked at that PR, our technically-unsound protocol typing was still going to work without error)